jupyterhub_saml_auth/handlers.py

```diff
@@ -0,0 +1,148 @@
+from jupyterhub.handlers import LoginHandler, BaseHandler, LogoutHandler
+from onelogin.saml2.auth import OneLogin_Saml2_Auth
+import tornado
+from tornado.log import app_log
+import tornado.web
+import os
+
+
+__all__ = [
+    'MetadataHandler',
+    'SamlLoginHandler',
+    'SamlLogoutHandler',
+    'ACSHandler'
+]
+
+
+def get_request(request):
+    dataDict = {}
+    for key in request.arguments:
+        dataDict[key] = request.arguments[key][0].decode('utf-8')
+
+    result = {
+        'https': 'on' if request.protocol == 'https' else 'off',
+        'http_host': tornado.httputil.split_host_and_port(request.host)[0],
+        'script_name': request.path,
+        'server_port': tornado.httputil.split_host_and_port(request.host)[1],
+        'get_data': dataDict,
+        'post_data': dataDict,
+        'query_string': request.query
+    }
+
+    return result
+
+
+class BaseHandlerMixin:
+
+    @property
+    def saml_settings_path(self):
+        return self._saml_settings_path
+
+    @saml_settings_path.setter
+    def saml_settings_path(self, path):
+        self._saml_settings_path_exists(path)
+        self._settings_files_exist(path)
+
+        self._saml_settings_path = path
+
+    def _saml_settings_path_exists(self, path):
+        # error checks
+        if not os.path.exists(path):
+            raise NotADirectoryError(f'Could not locate \
+                saml settings path at {path}')
+
+    def _settings_files_exist(self, path):
+        dir_contents = os.listdir(path)
+
+        files_to_check = ['settings.json', 'advanced_settings.json']
+
+        for f in files_to_check:
+            if f not in dir_contents:
+                raise FileNotFoundError(f'Could not locate {f} in saml \
+                    settings path. Path = {path}, contents \
+                    = {dir_contents}')
+
+
+class MetadataHandler(BaseHandler, BaseHandlerMixin):
+
+    def get(self):
+        request = get_request(self.request)
```

Comment:
Might be useful to name it `reformatted_request`/`auth_request` or something more descriptive to distinguish the original request and the new request

Reply:
Sounds like a better name